wallet/wallet.go

```diff
@@ -4,70 +4,358 @@ package wallet
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/multisig"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/utils"
+
+	"github.com/ava-labs/avalanchego/utils/formatting/address"
 
-	"github.com/ava-labs/avalanche-tooling-sdk-go/keychain"
-	"github.com/ava-labs/avalanchego/ids"
-	avagokeychain "github.com/ava-labs/avalanchego/utils/crypto/keychain"
 	"github.com/ava-labs/avalanchego/utils/set"
 	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
-	"github.com/ava-labs/avalanchego/wallet/subnet/primary"
 	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/tx"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/network"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/account"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/wallet/p-chain/txs"
+	"github.com/ava-labs/avalanchego/ids"
+	avagoTxs "github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary"
+	"github.com/ava-labs/subnet-evm/ethclient"
 )
 
-type Wallet struct {
+type ChainClients struct {
+	C *ethclient.Client // …/ext/bc/C/rpc
+	X string            // …/ext/bc/X
+	P string            // …/ext/bc/P
+}
+
+// LocalWallet represents a local wallet implementation
+type LocalWallet struct {
 	*primary.Wallet
-	Keychain keychain.Keychain
-	options  []common.Option
-	config   primary.WalletConfig
+	accounts []account.Account
+	clients  ChainClients
+}
+
+// Ensure LocalWallet implements Wallet interface
+var _ Wallet = (*LocalWallet)(nil)
+
+// NewLocalWallet creates a new local wallet
+func NewLocalWallet() (*LocalWallet, error) {
+	return &LocalWallet{
+		Wallet:   nil,
+		accounts: []account.Account{},
+		clients:  ChainClients{},
+	}, nil
 }
 
-func New(ctx context.Context, uri string, avaxKeychain avagokeychain.Keychain, config primary.WalletConfig) (Wallet, error) {
+func (w *LocalWallet) loadAccountIntoWallet(ctx context.Context, account account.Account, network network.Network) error {
 	wallet, err := primary.MakeWallet(
 		ctx,
-		uri,
-		avaxKeychain,
-		nil,
-		config,
+		network.Endpoint,
+		account.SoftKey.KeyChain(),
+		account.SoftKey.KeyChain(),
+		primary.WalletConfig{},
 	)
-	return Wallet{
-		Wallet: wallet,
-		Keychain: keychain.Keychain{
-			Keychain: avaxKeychain,
-		},
-		config: config,
-	}, err
-}
-
-// SecureWalletIsChangeOwner ensures that a fee paying address (wallet's keychain) will receive
-// the change UTXO and not a randomly selected auth key that may not be paying fees
-func (w *Wallet) SecureWalletIsChangeOwner() {
-	addrs := w.Addresses()
-	changeAddr := addrs[0]
-	// sets change to go to wallet addr (instead of any other subnet auth key)
-	changeOwner := &secp256k1fx.OutputOwners{
-		Threshold: 1,
-		Addrs:     []ids.ShortID{changeAddr},
+	if err != nil {
+		return err
+	}
+	w.Wallet = wallet
+	return nil
+}
+
+func (w *LocalWallet) Accounts() []account.Account {
```

Comment:
What is the difference between `Accounts()`, `GetAllAccounts()` and `ListAccounts()`?
Based on the interface, `ListAccounts` returns "managed" accounts, what does that mean?
What is the benefit of returning array of structs here in `ListAccounts()` (we don't use this function at all now)?

Reply:
Removed getallaccounts, keeping listaccounts